screen/contributors/src/main/java/com/ivy/contributors/IvyWalletRepositoryDataSource.kt

```diff
@@ -33,26 +38,61 @@ class IvyWalletRepositoryDataSource @Inject constructor(
         @SerialName("stargazers_count")
         val stars: Int,
         @SerialName("html_url")
-        val url: String
+        val url: String,
     )
 
     companion object {
         private const val CONTRIBUTORS_PER_PAGE = 100
+        private const val DISPLAY_ANONYMOUS_CONTRIBUTORS = true
+        private const val INITIAL_PAGE = 1
     }
 
-    suspend fun fetchContributors(): List<ContributorDto>? {
-        return try {
-            withContext(Dispatchers.IO) {
+    suspend fun fetchContributors(): Either<String, List<ContributorDto>> = either {
+        withContext(dispatchersProvider.io) {
+            try {
+                pagingSource().bind()
+            } catch (e: Exception) {
+                raise(e.message ?: "Unknown Error")
+            }
+        }
+    }
+
+    private suspend fun pagingSource(): Either<String, List<ContributorDto>> = either {
+        val contributorsSource = mutableListOf<ContributorDto>()
+        var currentPage: Int? = INITIAL_PAGE
+        while (currentPage != null) {
+            val contributorsResult = getContributorsFromRequest(currentPage)
+            contributorsResult.fold({ errorMessage ->
```

Comment:
Here, `.onLeft` and `onRight` would be more appropriate than `fold` because we do only side-effects and don't map the result.